store/localcachelayer/channel_layer.go

```diff
@@ -63,3 +81,79 @@ func (s LocalCacheChannelStore) GetMemberCountFromCache(channelId string) int64
 
 	return count
 }
+
+// ChannelCacheByName methods
+func (s LocalCacheChannelStore) GetByName(teamId string, name string, allowFromCache bool) (*model.Channel, *model.AppError) {
+	return s.getByName(teamId, name, false, allowFromCache)
+}
+func (s LocalCacheChannelStore) GetByNames(teamId string, names []string, allowFromCache bool) ([]*model.Channel, *model.AppError) {
+	var channels []*model.Channel
+
+	if allowFromCache {
+		var misses []string
+		visited := make(map[string]struct{})
+		for _, name := range names {
+			if _, ok := visited[name]; ok {
+				continue
+			}
+			visited[name] = struct{}{}
+			if cacheItem, ok := s.rootStore.channelByNameCache.Get(teamId + name); ok {
```

Comment:
same here, you can use the `s.rootStore.doStandardReadCache` to avoid increment the counters.